api/graphite.go

```diff
@@ -233,13 +239,14 @@ func (s *Server) renderMetrics(ctx *middleware.Context, request models.GraphiteR
 	}
 	plan, err := expr.NewPlan(exprs, fromUnix, toUnix, mdp, stable, opts)
 	if err != nil {
-		if fun, ok := err.(expr.ErrUnknownFunction); ok {
-			if request.NoProxy {
-				ctx.Error(http.StatusBadRequest, "localOnly requested, but the request cant be handled locally")
-				return
-			}
+		fun, isUnknownFunction := err.(expr.ErrUnknownFunction)
+		err := response.WrapError(err)
+		if err.HTTPStatusCode() == http.StatusBadRequest && !request.NoProxy {
+			log.Infof("Proxying to Graphite because of error: %s", err.Error())
 			s.proxyToGraphite(ctx)
-			proxyStats.Miss(string(fun))
+			if isUnknownFunction {
+				proxyStats.Miss(string(fun))
+			}
 			return
 		}
 		ctx.Error(http.StatusBadRequest, err.Error())
```

Comment:
we shouldn't always assume all these cases are bad request.
err.HTTPStatusCode() may return all kinds of different error codes.

Reply:
> we shouldn't always assume all these cases are bad request.
> err.HTTPStatusCode() may return all kinds of different error codes.

are you referring to the pre-existing code? `ctx.Error(http.StatusBadRequest, err.Error())`

would you rather have `ctx.Error(err.HTTPStatusCode(), err.Error())`